src/BinOp.chpl

```diff
@@ -1185,6 +1225,30 @@ module BinOp
         }
       var repMsg = "created %s".doFormat(st.attrib(rname));
       return new MsgTuple(repMsg, MsgType.NORMAL);
+    } else if (r.etype == bool && val.type == bool)  {
+      select op {
+          when "<<" {
+            if(val){
+              e.a = val:int << r.a:int;
+            }else{
+              e.a = val:int;
+            }
```

Comment:
In the else block, we know `val` is `False` and since casting `False` to int gives `0`,  the else block is setting `e.a = 0;`. `e.a` was initialized to be all zeroes, so this operation isn't necessary. We can just do 
```chapel 
if val {
  e.a = val:int << r.a:int;
}
```
with no else block. The same goes for the `">>"` case

Reply:
Ok, I made the change.